### ISSUE TITLE
build issues on alpine linux / musl libc

### ISSUE THREAD

**Issue by billzez** [jpbruyere/vkvg#79]
1. In cross_os.h, I had to change `#define isnanf __isnanf` to `#define isnanf isnan`.
2. vkvg_record_internal.c uses `va_arg`, but does not include `stdlib.h`.
3. tests/text segfaults if I build without fontconfig.  `ctx` is a null pointer in `_find_or_create_font_size` when trying to load the `mono` font.
4. If I build with freetype, I get:

```
[ 84%] Building C object tests/CMakeFiles/gradient2.dir/gradient2.c.o
In file included from /home/bill/vkvg/tests/../src/vkvg_context_internal.h:29,
                 from /home/bill/vkvg/tests/getarcstep.c:3:
/home/bill/vkvg/tests/../src/vkvg_fonts.h:29:11: fatal error: ft2build.h: No such file or directory
   29 |  #include <ft2build.h>
      |           ^~~~~~~~~~~~
compilation terminated.
```

**Comment by jpbruyere** [comment]
Can you give me the git commit you compiled? Thank you for your report...

**Comment by jpbruyere** [comment]
For your second point, I'm investigating, what is strange is that I don't even include stdarg.h...

**Comment by jpbruyere** [comment]
For the first point, what solution would you suggest? Maybe testing if running with musl lib with a macro? Is there a macro to check that?

**Comment by jpbruyere** [comment]
For the 3rd point, compiling without fontconfig, this should be solved in my current devel branch, I'll double checked that...I should merge in the next few days.

**Comment by billzez** [comment]
Thanks for the fast response.

I was testing latest master: https://github.com/jpbruyere/vkvg/commit/ce43dbcf6f4c8c5a29acaebdb425187958c9a63f

It's not uncommon that a missing include pops up when switching libc's.

`isnanf` is obsolete.  `isnan` works with both floats and doubles.  See `man fpclassify`

**Comment by jpbruyere** [comment]
So, the best solution would be to use ìsnan` in source, and double check for msvc only...

**Comment by jpbruyere** [comment]
just using `isnan` for android is also ok from my googling?

**Comment by Rubo3** [comment]
> vkvg_record_internal.c uses `va_arg`, but does not include `stdlib.h`.

@billzez did you mean `stdarg.h`?

**Comment by jpbruyere** [comment]
so having a clean stdarg include (double check if stdlib is always required) is required, thank you for your investigation :-)

**Comment by Rubo3** [comment]
Sure, I think it's generally better to include headers directly instead of relying on implicit inclusion.

**Comment by billzez** [comment]
Yeah, that was a typo, I meant `stdarg.h` not `stdlib.h`.

**Comment by jpbruyere** [comment]
@billzez : ok, so I've made also an error saying both are required, on windows that not stdlib, but stdio that is marked as required on doc: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/va-arg-va-copy-va-end-va-start?view=msvc-170

**Comment by jpbruyere** [comment]
@billzez : I've merged my devel branch into master, (4a6e6c2) but for the isnan change, I've made a branch [isnan_fix](https://github.com/jpbruyere/vkvg/tree/isnan_fix) and will wait to be able to compile on other platform (github action cancel win jobs, looks like a random problem on their side).

**Comment by billzez** [comment]
Thanks.  Unfortunately, there is a new issue now.  `backtrace()` and `execinfo.h` are glibc specific.  They should probably be guarded by `#ifdef __GLIBC__`

**Comment by billzez** [comment]
New test fails to build:

```
/home/bill/vkvg/tests/multithreading/multithreaded2.c: In function 'main':
/home/bill/vkvg/tests/multithreading/multithreaded2.c:61:2: error: 'vkvg_log_level' undeclared (first use in this function)
   61 |  vkvg_log_level = VKVG_LOG_ERR|VKVG_LOG_DEBUG;//|VKVG_LOG_INFO|VKVG_LOG_INFO_PATH|VKVG_LOG_DBG_ARRAYS|VKVG_LOG_FULL;
      |  ^~~~~~~~~~~~~~
/home/bill/vkvg/tests/multithreading/multithreaded2.c:61:2: note: each undeclared identifier is reported only once for each function it appears in
```

**Comment by jpbruyere** [comment]
fixed on master, thanks for the report. For the second bug (vkvg_log) the fix was on master, I rebased the isnan_fix branch.

**Comment by billzez** [comment]
Thanks.  I can confirm that all issues are resolved in the `isnan_fix` branch.

There are more issues for building android, I might take a stab at it later.

**Comment by jpbruyere** [comment]
Ok, thank you for your help, I've compile for android several weeks ago, but I needed some space on my hd, I will also recheck android build soon.
